src/NSwag.Generation/GenericResultWrapperTypes.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+using System.Linq;
+
+namespace NSwag.Generation
+{
+
+	internal static class GenericResultWrapperTypes
+	{
+		private readonly static String[] WrapperTypeNames =
+		{
+			"Task`1",
+			"ValueTask`1",
+			"JsonResult`1",
+			"ActionResult`1"
+		};
+
+		internal static bool IsGenericWrapperType(string typeName)
+			=> WrapperTypeNames.Contains( typeName );
```

Comment:
this is quite slow, probably better to just do 
``` 
=> typeName == "Task`1" || typeName == "ValueTask`1" || typeName== "JsonResult`1" || typeName == "ActionResult`1" 
```

Reply:
I have no idea what the difference is in practice - I tend to write simple stuff like this more for readability than marginal gains in execution time but I guess it all adds up. I did some searching and found this [Performance insight for Enumerable.Contains](https://learn.microsoft.com/en-us/visualstudio/profiling/perf-insights-contains) which points out the doing this with a HashSet<T> would make it O(1), however for this short list of only fours items I bet your code smaller and possibly faster, so I'm happy to make your suggested change.